tsfc/fem.py

```diff
@@ -699,4 +699,17 @@ def compile_ufl(expression, interior_facet=False, point_sum=False, **kwargs):
     result = map_expr_dags(context.translator, expressions)
     if point_sum:
         result = [gem.index_sum(expr, context.point_indices) for expr in result]
+    # Check that any indices from the PointSet have made it into the resulting expression
+    for i in range(len(result)):
+        expansion_indices = ()
+        for index in context.point_indices:
+            if index not in result[i].free_indices:
+                expansion_indices = expansion_indices + (index,)
```

Comment:
Strange way to write a set difference

`expand = tuple(i for i in indices if i not in free_indices)`

Reply:
Best way to learn how to do something the correct way: publicly show the wrong way and await correction :)